File: progtest-web/resources/oracles/Java/MaxMin/MaxMin.java
public class MaxMin {

	public static int maximum(int[] values) {
		int max = values[0];
		for(int i = 1; i < values.length; i++)
			if (max < values[i])
				max = values[i];
		return max;
	}

	public static int minimum(int[] values) {
		int min = values[0];
		for (int i = 1; i < values.length; i++)
			if (min > values[i])
				min = values[i];
		return min;
	}

	public static int[] maxmin(int[] values) throws IllegalArgumentException {
		if (values.length == 0)
			throw new IllegalArgumentException();
		else {
			int[] maxmin = new int[2];
			maxmin[0] = maximum(values);
			maxmin[1] = minimum(values);
			return maxmin;
		}
	}

}

File: progtest-web/resources/oracles/C/MaxMin/MaxMin.c
#include "MaxMin.h"

short int maximum(short int *values, unsigned char n) {
	short int max = values[0];
	unsigned char i;
	for (i = 1; i < n; i++)
		if (max < values[i])
			max = values[i];
	return max;
}

short int minimum(short int *values, unsigned char n) {
	short int min = values[0];
	unsigned char i;
	for (i = 1; i < n; i++)
		if (min > values[i])
			min = values[i];
	return min;
}

short int *maxmin(short int *values, unsigned char n) {
	short int *maxmin = (short int *) calloc(3, sizeof(short int));
	if (n > 0) {
		maxmin[0] = 1;
		maxmin[1] = maximum(values, n);
		maxmin[2] = minimum(values, n);
	} else {
		maxmin[0] = 0;
	}
	return maxmin;
}
